pkg/client/fake/client.go

```diff
@@ -949,7 +1083,10 @@ func (c *fakeClient) patch(obj client.Object, patch client.Patch, opts ...client
 	defer c.trackerWriteLock.Unlock()
 	oldObj, err := c.tracker.Get(gvr, accessor.GetNamespace(), accessor.GetName())
 	if err != nil {
-		return err
+		if patch.Type() != types.ApplyPatchType {
+			return err
+		}
+		oldObj = &unstructured.Unstructured{}
```

Comment:
Fine to ignore all errors in this case or could/should we only ignore certain errors? (like "not found")